lib/dialects/postgres/connection-manager.js

```diff
@@ -154,6 +154,9 @@ class ConnectionManager extends AbstractConnectionManager {
       }
 
       return new Promise((resolve, reject) => connection.query(query, (error, result) => error ? reject(error) : resolve(result))).then(result => {
+        if (Array.isArray(result)) {
+          result = result.pop();
```

Comment:
@felixfbecker Not sure how a test can reach this place. The better approach would be to increment the `pg`'s version to v7 for tests, so they all fail 😅 

Also, please consider changing to:

```js
...(results => {
const result = Array.isArray(results) ? results.pop() : results;
...
})
```

Assigning to a function parameter is not very pretty, or readable for that matter.